charts/tenantfilesystem/templates/composition.yaml

```diff
@@ -0,0 +1,808 @@
+          combine:
+            variables:
+              - fromFieldPath: "spec.product"
+              - fromFieldPath: "metadata.labels[crossplane.io/claim-namespace]"
+              - fromFieldPath: "metadata.labels[crossplane.io/claim-name]"
+            strategy: string
+            string:
+              fmt: "tenant-filesystem-%s-%s-%s-remote"
+          toFieldPath: "metadata.name"
+        - fromFieldPath: "spec.deletePolicy"
+          toFieldPath: "spec.deletePolicy"
+        - fromFieldPath: "metadata.name"
+          toFieldPath: "spec.forProvider.name"
+          transforms:
+            - type: string
+              string:
+                fmt: "%s-EFSDescribeMountTargetIAMPolicy"
+          policy:
+            fromFieldPath: Required
+        - type: CombineFromComposite
+          combine:
+            variables:
+              - fromFieldPath: "spec.region"
+              - fromFieldPath: "spec.accountID"
+              - fromFieldPath: "status.fileSystemID"
+            strategy: string
+            string:
+              fmt: |
+                {
+                  "Version": "2012-10-17",
+                  "Statement": [
+                    {
+                      "Sid" : "Stmt1DescribeMountTargets",
+                      "Effect": "Allow",
+                      "Action": [
+                        "elasticfilesystem:DescribeFileSystems",
+                        "elasticfilesystem:DescribeMountTargets",
+                        "elasticfilesystem:CreateAccessPoint"
+                      ],
+                      "Resource": "arn:aws:elasticfilesystem:%s:%s:file-system/%s"
+                    },
+                    {
+                      "Sid" : "Stmt2AdditionalEC2PermissionsToDescribeMountTarget",
+                      "Effect": "Allow",
+                      "Action": [
+                        "ec2:DescribeSubnets",
+                        "ec2:DescribeNetworkInterfaces"
+                      ],
+                      "Resource": "*"
+                    }
+                  ]
+                }
+          toFieldPath: "spec.forProvider.document"
+          policy:
+            fromFieldPath: Required
+        - type: ToCompositeFieldPath
+          fromFieldPath: "metadata.name"
+          toFieldPath: "status.policyNameRemote"
+
+    - name: role-policy-attach
+      base:
+        apiVersion: iam.aws.crossplane.io/v1beta1
+        kind: RolePolicyAttachment
+        metadata:
+          labels:
+            crossplane/xrd-kind: xtenantfilesystem.wadtfy.bmwgroup.net
+            crossplane/xrd-helm_chart_version: {{ .Chart.Version }}
+        spec:
+          deletionPolicy: Delete
+          forProvider:
+            policyArnRef:
+              name: ""
+            roleNameRef:
+              name: ""
+          providerConfigRef:
+            name: ""
+      patches:
+        - type: CombineFromComposite
+          combine:
+            variables:
+              - fromFieldPath: "spec.product"
```

Comment:
read from tenant EnvironmentConfig and not from spec of XRD

Reply:
changed